release: release notes for v0.18.0 (#345)
confidential-containers/community: releases/v0.18.0.md

# Release Notes for v0.18.0

> Release Date: January 22nd, 2025

Reviewer: 2026?

> This release is based on [3.25.0](https://github.com/kata-containers/kata-containers/releases/tag/3.23.0) of Kata Containers

Reviewer 2: ```suggestion
This release is based on [3.25.0](https://github.com/kata-containers/kata-containers/releases/tag/3.25.0) of Kata Containers
```

Trustee and the guest components use KBS protocol v0.4.0.

Please see the [quickstart guide](https://confidentialcontainers.org/docs/getting-started/) or [project documentation](https://confidentialcontainers.org/docs) for more information.

## Deprecation Notices
* Support for process-based confidential computing via enclave-cc has been removed.
* The CoCo operator has been deprecated. The project is now installed via a Helm [chart](https://github.com/confidential-containers/charts).
* The Cloud API Adaptor has dropped support for the experimental Secure Comms mode.

> * The Cloud API Adaptor has dropped support for the vSphere provider in the next release.

Reviewer 2: ```suggestion
* The Cloud API Adaptor will drop support for the vSphere provider in the next release.
```

* The Cloud API Adaptor is moving away from building guest images with packer in favor of mkosi.
  Support for packer images will be dropped in some future release.

> ## What's New

Reviewer 2: confidential guest kernel bumped to 6.18 LTS

Reviewer 2: TDX also moved to TDX hardened build of OVMF (202511 release)

* Trustee attestation policy can specify extensions for the attestation token.
* Basic validated identifiers added to default Trustee attestation policy. 
* AMD SNP verifier supports VCEK caching and pre-provisioning.
* Improved support for Trustee and guest-components non-x86 builds.
* Added image pull benchmarking tool to image-rs.
* AAEL (eventlog) now covers pull image events.
* Image signature configuration can be specified directly in CDH config file (and init-data).
* Improvements to image-rs whiteout handling.
* TPM attester improvements.
* Improved error reporting in image-rs.

## Hardware Support

Attestation is supported and tested on three platforms: Intel TDX, AMD SEV-SNP, and IBM SEL.
Not all features have been tested on every platform, but those based on attestation
are expected to work on the platforms above.

Make sure your host platform is compatible with the hypervisor and guest kernel
provisioned by CoCo.

This release has been tested on the following stacks:

### AMD SEV-SNP

* Processor: AMD EPYC 7413
* Kernel: 6.16.1
* OS: Ubuntu 22.04.5 LTS
* k8s: v1.33.0 (Kubeadm)
* Kustomize: v5.6.0
* Containerd: v1.7.27

### Intel TDX

> * Kernel: [6.8.0-1028-intel](https://git.launchpad.net/~kobuk-team/ubuntu/+source/linux-intel/tree/?h=noble-main-next)

Reviewer 2: ```suggestion
* Kernel: [6.8.0-1028-intel](https://git.launchpad.net/~kobuk-team/ubuntu/+source/linux-intel/tree/?h=noble-main-next)*
```

I'd like to propose a footnote in this section to say we drop the support of these kobuk builds in the next release and move to use 24.04.4 HWE kernel and  runtime handler configuration that uses QEMU from the CoCo payload/kata-deploy.

* OS: Ubuntu 24.04 LTS

> * k8s: v1.34.2 (Kubeadm)

Reviewer 2: ```suggestion
* k8s: v1.35.0 (Kubeadm)
```
